bokeh/core/has_props.py

```diff
@@ -29,16 +29,36 @@
 
 # Standard library imports
 import difflib
-from functools import lru_cache
 from typing import (
+    TYPE_CHECKING,
     Any,
+    Callable,
     Dict,
+    Iterable,
+    List,
+    NoReturn,
     Optional,
     Set,
+    Type,
+    TypeVar,
     Union,
+    overload,
 )
 from warnings import warn
 
+# External imports
+from typing_extensions import Literal
+
+if TYPE_CHECKING:
+    F = TypeVar("F", bound=Callable[..., Any])
+    def lru_cache(arg: Optional[int]) -> Callable[[F], F]: ...
+else:
+    from functools import lru_cache
+
+if TYPE_CHECKING:
+    from .property.dataspec import DataSpec
+    from .property.bases import Property
```

Comment:
`functools` is stdlib (not external) and `.property` is  Bokeh (not external). Maybe move these under Globals and constants?